votes/0625-new-triage-method-for-cherrypick.md

```diff
@@ -0,0 +1,54 @@
+
+[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgICBBW25ldyBwdWxsIHJlcXVlc3RdIC0tPiBCe2hhcyBjb3JyZXNwb25kaW5nIGlzc3VlfVxuICAgIEIgLS0-IHxZZXN8IEN7aXNzdWUgaGFzIHR5cGUveHggbGFiZWx9XG4gICAgQiAtLT4gfE5vfCBEW2NoZWNrZXIgZmFpbHNdXG4gICAgQyAtLT4gfFllc3wgRXtpc3N1ZSBoYXMgdHlwZS9idWcgbGFiZWx9XG4gICAgQyAtLT4gfE5vfCBEXG4gICAgRSAtLT4gfFllc3wgR3tpc3N1ZSBoYXMgc2V2ZXJpdHkveHggbGFiZWx9XG4gICAgRSAtLT4gfE5vfCBGW2NoZWNrZXIgcGFzc2VzXVxuICAgIEcgLS0-IHxZZXN8IEh7aXNzdWUgaGFzIHNldmVyaXR5L2NyaXRpY2FsIG9yIHNldmVyaXR5L21ham9yIGxhYmVsfVxuICAgIEcgLS0-IHxOb3wgRFxuICAgIEggLS0-IHxZZXN8IEl7aXNzdWUgaGFzIGFueSBtYXktYWZmZWN0LXgueCBsYWJlbH1cbiAgICBIIC0tPiB8Tm98IEZcbiAgICBJIC0tPiB8WWVzfCBEXG4gICAgSSAtLT4gfE5vfCBGXG4gICAgRiAtLT4gSltib3QgYWRkcyBuZWVkcy1jaGVycnktcGljay14LnggbGFiZWxzIGFjY29yZGluZyB0byBpc3N1ZSBhZmZlY3RzLXgueCBsYWJlbHNdXG4gICAgSiAtLT4gS1tib3QgY3JlYXRlcyBjaGVycnktcGljayBwdWxsIHJlcXVlc3RzIGFmdGVyIG1lcmdlXSIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2UsImF1dG9TeW5jIjp0cnVlLCJ1cGRhdGVEaWFncmFtIjpmYWxzZX0)](https://mermaid-js.github.io/mermaid-live-editor/edit#eyJjb2RlIjoiZ3JhcGggVERcbiAgICBBW25ldyBwdWxsIHJlcXVlc3RdIC0tPiBCe2hhcyBjb3JyZXNwb25kaW5nIGlzc3VlfVxuICAgIEIgLS0-IHxZZXN8IEN7aXNzdWUgaGFzIHR5cGUveHggbGFiZWx9XG4gICAgQiAtLT4gfE5vfCBEW2NoZWNrZXIgZmFpbHNdXG4gICAgQyAtLT4gfFllc3wgRXtpc3N1ZSBoYXMgdHlwZS9idWcgbGFiZWx9XG4gICAgQyAtLT4gfE5vfCBEXG4gICAgRSAtLT4gfFllc3wgR3tpc3N1ZSBoYXMgc2V2ZXJpdHkveHggbGFiZWx9XG4gICAgRSAtLT4gfE5vfCBGW2NoZWNrZXIgcGFzc2VzXVxuICAgIEcgLS0-IHxZZXN8IEh7aXNzdWUgaGFzIHNldmVyaXR5L2NyaXRpY2FsIG9yIHNldmVyaXR5L21ham9yIGxhYmVsfVxuICAgIEcgLS0-IHxOb3wgRFxuICAgIEggLS0-IHxZZXN8IEl7aXNzdWUgaGFzIGFueSBtYXktYWZmZWN0LXgueCBsYWJlbH1cbiAgICBIIC0tPiB8Tm98IEZcbiAgICBJIC0tPiB8WWVzfCBEXG4gICAgSSAtLT4gfE5vfCBGXG4gICAgRiAtLT4gSltib3QgYWRkcyBuZWVkcy1jaGVycnktcGljay14LnggbGFiZWxzIGFjY29yZGluZyB0byBpc3N1ZSBhZmZlY3RzLXgueCBsYWJlbHNdXG4gICAgSiAtLT4gS1tib3QgY3JlYXRlcyBjaGVycnktcGljayBwdWxsIHJlcXVlc3RzIGFmdGVyIG1lcmdlXSIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)
+
+By the way, there is also a `needs-more-info` label. Every time an issue is closed the bot will add it and leave a comment to remind the creators and reviewers to do cherrypick. But actually it doesn't do much help and becomes a little bit cumbersome. Currently there are more than 600 issue with the label not yet handled. And its name is confusing, some developers use it to ask for more information. After the new workflow, we don't need it and will delete it.
+
+## Affected Repositories
```

Comment:
some repositories are merged into below ones, but old release versions are still maintained in the original repositories, e.g. https://github.com/pingcap/dumpling/releases 

Do they need a special logic?

Reply:
From my point of view, the only problem for such cases is there will be more cherry-pick PRs than needed, right? A issue might have multiple `affects-x.x` labels but some versions are still in the old repo. I am not sure what should we do in such circumstances. What's your suggestion?

Reply:
My point of view is that the goal of the modification of the triage process is to minimize the risk of missing cherrypicks to an affected released branch. 

I don't know how such case is handled currently, but the modification itself doesn't bring any more trouble to the process, since the `affects-x.y` label will be correct no matter where the code is.

As for cherrypicking, if the RFC is implemented, as @zhangyangyu said, the bot will create cherrypicks on TiDB that are invalid. I think it is okay for the following reasons:

1. since the file organization has changed, it will be very hard and error-prone to create cherrypick PRs to the original repositories automatically, therefore I think it inevitably requires contributor's intervention to manually create cherrypick PRs.
2. due to the above reason, the bot can be configured to do not cherrypick on certain versions where the PR fixes bugs in code that recently migrated to TiDB, however I also doubt that the logic for the configuration and identification can be very hard
3. while the bot created invalid cherrypicks, the contributor can simply close it in such cases, and it can be a clue to the contributor: such cherrypick is not valid, please manually cherrypick it to the original repository, it is welcomed or should be enforced that the manual cherrypick PR reference the invalid cherrypick PR so that we can find the correct cherrypick through the closed invalid cherrpick.
4. as for release triage, we inevitably need to put more manual check for such cases (no matter how we improve the process, the root of the trouble exists from the first day we migrate repos to TiDB), with the invalid cherrypicks closed, it is a good hint to the release manager that such cherrypick needs more careful treatment.

Therefore, my conclusion is, we don't need modification to the bot's process, but only need special treatment manually for such cases.